### ISSUE TITLE
Initializing plugin while in GOTR not working

### ISSUE THREAD

**Issue by GitVexy** [Infinitay/essence-pouch-degrading#32]
Just got this plugin and have tinkered around with it trying to get the decay state to be accurate. 

Repairing while not decayed does not initialize, and neither does repairing while decayed. There are also definitely 12 essence in the leftmost pouch, but after repairing, it defaulted to 9 still, and does not update on checking.
![image](https://github.com/user-attachments/assets/1c05169d-d2a9-4f39-bc2f-af5528772885)

**Comment by Infinitay** [comment]
How are you repairing the pouches? Are you contacting the Dark Mage via NPC contact or are you using the GOTR NPC Cordelia? I was just about to head off to bed but I will take a look at this first thing in the morning.

If you want to try one more thing, could you try to reset the plugin via the settings, turning the plugin on/off, and try to repair it once more? Thank you.

![image](https://github.com/user-attachments/assets/23a89323-ef25-484e-83d2-9c6a52a0a283)

> There are also definitely 12 essence in the leftmost pouch, but after repairing, it defaulted to 9 still

That's odd. I figured it could have been an issue with the logic where it would think unknown pouch states can only fill a maximum of the degraded state (12 regular giant pouch -> 9 degraded giant pouch). However, I just tried it myself in game and it filled my pouch appropriately to 12 instead of 9. I'll take a look into this further and see if I can reproduce it. Sorry for the inconvenience. **In the future, could you try to check the pouch if it has an incorrect stored count? That should fix it.**


EDIT: Oh, by any chance, did you start the plugin while your giant pouch already had some essence in it, and proceed to fill it before checking it?

**Comment by GitVexy** [comment]
Thank you for looking into this so quickly, and please do get some rest.

I am repairing using cordelia. Further repairs did not fix the issue, and checking the pouch while it incorrectly stated 9 essence did not update it to 12, even though it said it contained 12 in the chatbox.

I also started the plugin completely fresh before starting a game of GOTR. I simply:
Logged in,
got the plugin while waiting for the last game to finish,
noticed it said "Repair",
repaired with Cordelia, which didn't update them,
played until the pouch decayed (correct essence tracking the entire time),
repaired with Cordelia, still not updating decay numbers,
kept playing, with correct numbers still,
after another decay, I repaired again, and then the tracking bug started.

It did not resolve itself through natural means, so I defaulted back to the other plugin in the meantime. I did not think to reset at that point, unfortunately.

Let me know if there are other things I can try. I will report back if I return to GOTR today.

**Comment by Infinitay** [comment]
tldr (not really)
- The **non-colossal** pouches not updating when you repair via Cordelia was because of a flag I set whenever a player logs in, but you were already logged in
	- **To resolve this in the meantime you can log out and then back with the plugin still on**
	- Considering either removing the flag entirely as I only added it thinking there would be micro-performance _gains_ avoiding checking every single dialog. Current plan is to also set it when you first start the plugin up which would mitigate this.
- I think the reason your stored essence count was off was because **you may have already had some essence in another pouch before the plugin was installed** and filled it fully after installing the plugin
	- I think I found the cause and I'll be implementing at least a mitigation if not a fix for it
- I'm not sure why you were having issues with checking pouches not updated to the correct stored amount
	- Are you using any plugins that modifies your chat?
	- Did you modify any in-game chat options such as coloring maybe? I doubt this would cause issues but I want to recreate your setup and see if there's any issues potentially

### If you'd like to give the plugin one more try in the current version until I can push out the changes and merged into the Plugin Hub: Log out of the OSRS and reset the plugin. Log back in and check your pouches and attempt to repair again.

Sorry for getting back to you late. I could recreate your issue with the wrong stored count, but you did make me realize a few things I had wrong. I'm sorry about the inconvenience. The first is that the initialization is missing some steps and could cause issues such as when resetting and not "refreshing" your inventory (eg. dropping and picking back up or moving something around). I guess I never noticed it because I had an existing config, and when I did reset the plugin I logged back in and out. I already implemented some changes and have been testing them. 

As for the repair issue, I found out last night that the repair dialogs and options change when you use regular vs colossal pouches (see #30). Unfortunately the beta worlds went down when I woke up yesterday. I had made sure to keep my regular pouches saved to the beta world and on the main game I upgraded to a colossal pouch. I finally acquired a new set of regular pouches and I'll be going through the dialogues once again to make sure everything is working. **For your issue regarding not being able to repair in particular via Cordelia:** the issue is that it think you don't have the ability to repair unlocked. On a bigger scale, it's because, again, I didn't consider the other variables for the plugin when you install it for the first time. I'm setting the flag whenever the game returns the VarbitChanged event; however, because you're already logged in and playing, the event never gets triggered. I didn't realize RuneLite doesn't _resend_ events. Sorry about that.

As for the stored essence bug, I managed to recreate a similar issue by doing the following:
1. I filled a pouch partially without the plugin
2. I installed the plugin and didn't check the pouches
3. I filled that pouch again fully and had left over essence
4. I used the remaining essence to fill another pouch
5. That last pouch I filled was filled, and I had essence left over still, but the stored count was wrong and less than it should have been

However, I did not experience your issue where checking didn't fix it. When I checked the pouch it properly updated. My only guess for this was that my manual tracking of the essence in the inventory was bugged and therefore sub-sequent fills became bugged too, especially since checking your pouches wasn't updating the count to mitigate this.

**Comment by GitVexy** [comment]
Thank you for all the effort you've put into this. I must say, I don't really have the time to help test this with you further, and safe to say that replicating my setup would require a lot of work, as I have several hundred of them installed (I check for new ones daily, and try most of them). I have many plugins that change the chat and its colors, but the game chats (the checks for example) remain unaltered for checking the pouch, both color and functionality.

Regarding this: **For your issue regarding not being able to repair in particular via Cordelia: the issue is that it think you don't have the ability to repair unlocked.**

I obviously do have the ability to repair unlocked, as I repaired my pouch multiple times. I think you'd do well in changing the way the plugin checks that flag. I realize now that relogging might've solved the issue, but it's always best to completely idiot-proof the mechanism, and I'm sure you'll figure it out; you seem quite capable.

For my sake, I think this issue has been resolved. I haven't gone back to GOTR yet, but I will be sure to report any further issues, should I find any.

Thanks again. And sorry I can't be of further assistance; I'm not a developer, and I can't take on more projects than I've got right now. If I did choose to help you, I'd hyperfocus on it until the problem was solved, and I can't allow myself to do that. I hope you understand.

**Comment by Infinitay** [comment]
>  I think you'd do well in changing the way the plugin checks that flag. I realize now that relogging might've solved the issue, but it's always best to completely idiot-proof the mechanism, and I'm sure you'll figure it out; you seem quite capable.

Yea I did this already and just testing it out.

Thanks for pointing out the errors with the plugin currently, and don't worry about testing or contributing. Testing should be my job and I'm not expecting anyone to dedicate their time on this spaghetti code.

**Comment by Infinitay** [comment]
Sorry for the long delay @devvExit  I kept finding new issues and adding new features and I wanted to bundle it all into one release. I submitted the update request to the Plugin Hub so hopefully it'll be reviewed when a maintainer gets the chance to do so, and accepted.

Plugin Hub Update Status: https://github.com/runelite/plugin-hub/pull/6925

I'll go ahead and close this issue for now. If you face any more problems please let me know and thanks for bring this to my attention.
